internal/tiltfile/tiltfile_test.go

```diff
@@ -2588,13 +2589,13 @@ func TestDefaultRegistryAtEndOfTiltfile(t *testing.T) {
 	f.file("Tiltfile", `
 docker_build('gcr.io/foo', 'foo')
 k8s_yaml('foo.yaml')
-default_registry('bar.com')
+default_registry('bar.com/baz')
```

Comment:
why did this change? is bar.com not a valid registry?

Reply:
It isn't, see all of the tests here:

https://github.com/moby/moby/blob/master/registry/registry_test.go#L309

Reply:
it sure looks like it is? https://github.com/moby/moby/blob/master/registry/registry_test.go#L415

or https://docs.docker.com/registry/ has
`docker push localhost:5000/myfirstimage`

IIUC this change is validating the registry name by parsing it as an image name, which probably isn't what we want. Maybe we could append a dummy image name?

Reply:
ah yeah, okay. I wonder if there's a function that validates a repository...

Reply:
ya, i think you want to use the DomainRegexp here: https://godoc.org/github.com/docker/distribution/reference#pkg-variables

Reply:
actually, scratch that, because i guess you can also have example.com/my-bucket be a registry?

Reply:
I think it just has to contain a `.` or a `:` to avoid the error: https://github.com/windmilleng/tilt/blob/6297a6f3aeb93ee5a8dcaaed5921acfe67455060/vendor/github.com/docker/distribution/reference/normalize.go#L64-L63

but even then it's complicated:
1. it might [fail later if it the resulting name is too long](https://github.com/windmilleng/tilt/blob/6297a6f3aeb93ee5a8dcaaed5921acfe67455060/vendor/github.com/docker/distribution/reference/reference.go#L203)
2. "rose-quartz" might be a valid registry name (I couldn't find the rules for valid registry names) that we should allow, and the issue is we're assembling combining it with the image name in an invalid way
3. regardless of the other stuff, it seems like we have a bug somewhere that we're logging this error without any context (resource or image name), and maybe if we just fix that, users can figure it out?

Reply:
> and maybe if we just fix that, users can figure it out?

nm. it's easy to change the error message, to, e.g.
`error parsing rose-quartz/vigoda after applying default registry rose-quartz: repository name must be canonical`

but I'd still have no idea what that means